lib/hanami/console/plugins/slice_readers.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Hanami
+  module Console
+    module Plugins
+      class SliceReaders < Module
+        def initialize(ctx)
+          ctx.application.slices.each do |slice|
+            define_method(slice.name) do
+              SliceDelegator.new(slice)
+            end
+          end
+        end
+
+        class SliceDelegator < SimpleDelegator
```

Comment:
@timriley I'm not sure if this is valid for this case, but for proxy objects it's often advisable to use `BasicObject` as superclass to reduce to the minimum the clash between methods available to `Object` subclasses vs `BasicObject`.

If you think it's the case, please consider to use `Hanami::Utils::BasicObject`, which has some few extra features that are worth to have. On top of all: `#inspect` and `"pp"` compat. Given we're talking about an interactive console, those features are really valuable.

Reply:
@jodosha Thanks for the heads up about `Hanami::Utils::BasicObject`, that sounds like a handy convenience.

However, in this case, I think we actually want the `SimpleDelegator`-provided behaviour, passing through all known method calls to the underlying slice object, just with that one added enhancement via the `#method_missing` implementation (though I'm sure there are a few other shortcuts we can add in addition to this, later on).